onnx/defs/schema.h

```diff
@@ -667,4 +669,39 @@ class OpSchemaRegistry final {
 // Helper function
 size_t ReplaceAll(std::string& s, const char* from, const char* to);
 
+typedef Common::Status (*BuildFunction)(std::shared_ptr<FunctionProto>*);
+
+class FunctionBuilder {
+ public:
+  FunctionBuilder& SetDomain(const std::string& domain);
+  const std::string& GetDomain() const;
+  FunctionBuilder& SetBuildFunction(BuildFunction build_func);
+
+  BuildFunction GetBuildFunction() const;
+
+ private:
+  std::string domain_;
+  BuildFunction build_func_;
+};
+
+class FunctionBuilderRegistry {
+ public:
+  FunctionBuilderRegistry() = default;
+
+  void Register(const FunctionBuilder& function_builder);
+
+  // Get functions for specific domain.
+  Status GetFunctions(
+      const std::string& domain,
+      /*out*/
+      std::multimap<std::string, std::shared_ptr<FunctionProto>>* function_set)
+      const;
+
+  static FunctionBuilderRegistry& OnnxInstance();
```

Comment:
nit: it's already inside the onnx namespace, so IMO no need to have "Onnx" in its name